File: stringCompression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        
        string a = "";
        
        if(chars.size() == 1)
        {
            char x = chars[0];
            chars.clear();
            chars.push_back(x);
            return 1;
        }
        
        int p1 = 0, p2 = 1, n = chars.size(), c = 1;
        
        
        while(p2 < n)
        {
            if(chars[p1] == chars[p2])
            {
                c++;
                p2++;
            }
            else
            {
                a += chars[p1];
                if(c != 1)
                    a += to_string(c);
                p1 = p2;
                p2++;
                c = 1;
            }
        }
        
        a += chars[p1];
        if(c != 1)
            a += to_string(c);
        
        
        chars.clear();
        
        for(char s : a){
            chars.push_back(s);
        }
        return chars.size();
    }
};

File: progress_sheet/stringCompression.py
class Solution:
    
    # Overwrite the number of occurenece of a character
    def writeOccurence(self, chars, write, count):
        
        if count != 1:
            count = str(count)

            for char in count:
                chars[write] = char
                write += 1

        return write

    def compress(self, chars: List[str]) -> int:

        size = len(chars)
        left = 0
        write = 0

        # append empty character to handle the last element
        chars.append(" ")

        if size == 1:
            return 1

        # when a new character is found write the occurence and go to the next character
        for right in range(size + 1):
            
            if chars[left] != chars[right]:

                chars[write] = chars[left]
                count = right - left
                write += 1

                # write the occurence only when it is greater than 1
                write = self.writeOccurence(chars, write, count)
                    
                left = right
        
        return write
